### DISCUSSION TITLE
[@typescript/react] [@types/react] React v18 update

### DISCUSSION THREAD

**Discussion by Iipal** [DefinitelyTyped/DefinitelyTyped#59659] [Issues with a @types package]
at least there is no `useTransition` and `useId` hooks typed

```
react-dom: ^18.0.0
@types/react: ^17.0.43
```

![image](https://user-images.githubusercontent.com/31858171/161390831-cd6bc4e8-0029-4126-b7da-a8d8227cceda.png)

[a comment by an automated account is not shown]

**Comment by Maskedman99** [comment]
Hi @Iipal were you able to fix the warning ``You are importing createRoot from "react-dom" which is not supported. You should instead import it from "react-dom/client".``. 
But when trying to use createRoot Typescript throws error ``Property 'createRoot' does not exist on type 'typeof import("/home/xxx/node_modules/@types/react-dom/index")'``

**Reply by Iipal** [reply]
@Maskedman99 Hi, just follow instructions on this [upgrade-to-v18 guide](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html). This works for me:
```ts
import { createRoot } from "react-dom/client";
import App from "./App";

const container = document.getElementById("root");
const root = createRoot(container as NonNullable<HTMLElement>);

root.render(<App />);
```

**Reply by Iipal** [reply]
If you still have troubles just re-install node_modules ;) Because this method above works perfectly.
